<?xml version="1.0"?>
<launch>
  <!-- DBW system -->
  <arg name="sys" default="true"/>
  <arg name="dataspeed_can" default="true" />
  <arg name="socketcan" default="false" />
  <arg name="socketcan_dev" default="can0" />

  <group if="$(var sys)">
    <include file="$(find-pkg-share dbw_ford_can)/launch/dbw.launch.xml" >
      <arg name="dataspeed_can" value="$(var dataspeed_can)" />
      <arg name="socketcan" value="$(var socketcan)" />
      <arg name="socketcan_dev" value="$(var socketcan_dev)" />
    </include>
  </group>

  <!-- Rviz -->
  <arg name="rviz" default="false"/>
  <group if="$(var rviz)">
    <include file="$(find-pkg-share dbw_ford_description)/launch/rviz2.launch.xml" />
  </group>

  <!-- Joystick demo -->
  <arg name="brake" default="true"/>
  <arg name="throttle" default="true"/>
  <arg name="steer" default="true"/>
  <arg name="shift" default="true"/>
  <arg name="signal" default="true"/>
  <arg name="brake_gain" default="1.0"/>
  <arg name="throttle_gain" default="1.0"/>
  <arg name="ignore" default="false"/>
  <arg name="enable" default="true"/>
  <arg name="count" default="false"/>
  <arg name="strq" default="false"/>
  <arg name="svel" default="0.0"/>
  <node namespace="vehicle" pkg="dbw_ford_joystick_demo" exec="joy_demo" output="screen" >
    <param name="brake" value="$(var brake)" /> <!-- Send brake commands -->
    <param name="throttle" value="$(var throttle)" /> <!-- Send throttle commands -->
    <param name="steer" value="$(var steer)" /> <!-- Send steering commands -->
    <param name="shift" value="$(var shift)" /> <!-- Send shift commands -->
    <param name="signal" value="$(var signal)" /> <!-- Send turn signal commands -->
    <param name="brake_gain" value="$(var brake_gain)" /> <!-- Adjust brake value -->
    <param name="throttle_gain" value="$(var throttle_gain)" /> <!-- Adjust throttle value -->
    <param name="ignore" value="$(var ignore)" /> <!-- Ignore driver overrides -->
    <param name="enable" value="$(var enable)" /> <!-- Enable buttons -->
    <param name="count" value="$(var count)" /> <!-- Watchdog counter -->
    <param name="strq" value="$(var strq)" /> <!-- Steering torque command (otherwise angle) -->
    <param name="svel" value="$(var svel)" /> <!-- Steering velocity limit -->
  </node>

  <!-- Joystick driver -->
  <arg name="dev" default="/dev/input/js0"/>
  <node pkg="joy" exec="joy_node" output="screen" >
    <param name="dev" value="$(var dev)" />
    <param name="deadzone" value="0.0" />
    <param name="autorepeat_rate" value="50.0" />
  </node>

</launch>
